### PR TITLE
apps: enabled psa for dex and cert-manager

### LINKED ISSUE #1447: Replace PSPs for cert-manager and dex
**Is your feature request related to a problem? Please describe.**
We have now investigated both PSA and gatekeeper as a replacement for PSPs. We have decided that we want to try to replace it with a combination of both.

**Describe the solution you'd like**
The idea is to use PSA for coarse grained control and gatekeeper for fine grained control. The PSA used in a namespace have to match the most privileged pod in that namespace, then we use gatekeeper to lock everything down as much as possible.

Use https://github.com/elastisys/compliantkubernetes-apps/pull/1444 as a basis.

**Describe alternatives you've considered**

**Additional context**

**Definition of done:**

- We have replaced all PSPs for cert-manager and dex.

### PR DESCRIPTION
**Pull request by crssnd**
**What this PR does / why we need it**: to enabled PSA for dex and cert-manager

**Which issue this PR fixes** *(use the format `fixes #<issue number>(, fixes #<issue_number>, ...)` to automatically close the issue when PR gets merged)*: fixes #1447 

**Special notes for reviewer**:
- dex:
   - changes: I added the necessary securitycontext setting to match the restricted policy
   - tested: delete and re-install the chart, logs, login using dex
- cert-manager
   - changes: nothing all the setting were add in the upstream chart
   - tested: delete and re-install the chart, logs, force renew a certificate and check that it worked
   - tbd: there are some situations when the webhook component might need hostNetwork and/or hostPorts, but as we do not use it at the moment I didn't switch to gatekeeper contrains

**Checklist:**

- [x] Added relevant notes to [WIP-CHANGELOG.md](https://github.com/elastisys/compliantkubernetes-apps/blob/main/WIP-CHANGELOG.md)
- [x] Proper commit message prefix on all commits
- [ ] Updated the [public facing documentation](https://github.com/elastisys/compliantkubernetes)
- Is this changeset backwards compatible for existing clusters? Applying:
    - [x] is completely transparent, will not impact the workload in any way.
    - [ ] requires running a migration script.
    - [ ] will create noticeable cluster degradation.
          E.g. logs or metrics are not being collected or Kubernetes API server
          will not be responding while upgrading.
    - [ ] requires draining and/or replacing nodes.
    - [ ] will change any APIs.
          E.g. removes or changes any CK8S config options or Kubernetes APIs.
    - [ ] will break the cluster.
          I.e. full cluster migration is required.
- Chart checklist (pick exactly one):
    - [x] I upgraded no Chart.
    - [ ] I upgraded a Chart and determined that no migration steps are needed.
    - [ ] I upgraded a Chart and added [migration steps](https://github.com/elastisys/compliantkubernetes-apps/blob/main/migration).

### CODE REVIEW

**Comment by robinAwallace** [comment]
I would like to say that we can disable the PSP now, as long as PSA and kubernetes PSP can run at the same time. 
The other option is to put it behind a flag that we can disable/remove in the feature.

File: WIP-CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@
 ### Added
 
 - Add PrometheusRule to alert for dropped packets to/from workloads.
+- Enabled Pod Security Admission for dex and dert-manager
```
Comment (OlleLarsson) [inline WIP-CHANGELOG.md:13]:
```suggestion
- Enabled Pod Security Admission for `dex` and `cert-manager`
```

**Comment by crssnd** [comment]
> I would like to say that we can disable the PSP now, as long as PSA and kubernetes PSP can run at the same time. The other option is to put it behind a flag that we can disable/remove in the feature.

I tested on a 1.24.6 kubernetes cluster and seems that PSA and PSP are running at the same time.
I removed the PSP for cert-manager and didntt notice any issues.

**Review by viktor-f — APPROVED** [review]
I think this looks good

**Comment by robinAwallace** [comment]
> do we want to pin the policy to a specific version?

No, i would go with the default PSA version. Which is the current kubernetes version running in the cluster

**Review by robinAwallace — APPROVED** [review]

File: helmfile/values/cert-manager.yaml.gotmpl [hunk 2]
```diff
@@ -17,7 +17,3 @@ cainjector:
   tolerations:  {{- toYaml .Values.certmanager.cainjector.tolerations | nindent 4 }}
 
 installCRDs: true
```
Comment (OlleLarsson) [inline helmfile/values/cert-manager.yaml.gotmpl:19]:
```suggestion
installCRDs: true

global:
  podSecurityPolicy:
    enabled: {{ .Values.kubernetesPSP }}
```

**Review by aarnq — APPROVED** [review]

**Review by OlleLarsson — APPROVED** [review]

**Pull request merged**